### ISSUE TITLE
Support multiple items with the same name when they have cfg attributes

### ISSUE THREAD

**Issue by eqrion** [eqrion/cbindgen#52]
Example:
```rust
#[cfg(target_os = "windows")]
#[repr(C)]
pub struct Font {
  data: i32,
}
 
#[cfg(target_os = "macos")]
#[repr(C)]
pub struct Font {
  data: char,
}
```
In this case, it is okay to have two items with the same name, because they will have #ifdefs to prevent them from both being compiled.

**Comment by eqrion** [comment]
cc @lsalzman

**Comment by eqrion** [comment]
This was implemented for `struct` and `enum` in 5f4d517883d3cab62ee4147515ed8befb403e645. The implementation was kind of rough, so I'll close this once it works for all items.
